test/integration/consul-container/libs/consul-node/consul-container.go

```diff
@@ -0,0 +1,130 @@
+package consul_node
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/docker/docker/pkg/ioutils"
+
+	"github.com/hashicorp/consul/integration/consul-container/libs/utils"
+
+	"github.com/testcontainers/testcontainers-go"
+
+	"github.com/hashicorp/consul/api"
+)
+
+const bootLogLine = "Consul agent running"
+const disableRYUKEnv = "TESTCONTAINERS_RYUK_DISABLED"
+
+// consulContainerNode implement a ConsulNode
+// it instantiate Consul as a container
+type consulContainerNode struct {
+	ctx       context.Context
+	client    *api.Client
+	container testcontainers.Container
+	ip        string
+	port      int
+}
+
+// NewConsulContainer create a ConsulNode implemented as a consulContainerNode
+func NewConsulContainer(ctx context.Context, config Config) (ConsulNode, error) {
+
+	name := utils.RandName("consul-")
+	ctx = context.WithValue(ctx, "name", name)
+	tmpDir, err := ioutils.TempDir("", name)
+	if err != nil {
+		return nil, err
+	}
+	err = os.Chmod(tmpDir, 0777)
+	if err != nil {
+		return nil, err
+	}
+	err = os.Mkdir(tmpDir+"/config", 0777)
+	if err != nil {
+		return nil, err
+	}
+	configFile := tmpDir + "/config/config.hcl"
+	err = os.WriteFile(configFile, []byte(config.HCL), 0644)
+	if err != nil {
+		return nil, err
+	}
+	skipReaper := isRYUKDisabled()
```

Comment:
Why does RYUK mean? I see that it controls whether the lib setups a reaper for this container but feel like I am missing some meaning here.

Reply:
Same! A quick Google brought up [this character](https://en.wikipedia.org/wiki/Ryuk_(Death_Note))? 💀😅 Could we just call it `skipReaper` instead?

Reply:
https://github.com/testcontainers/moby-ryuk

Reply:
Gotcha, thanks! Mind putting that link in a doc comment on the method, for future readers?